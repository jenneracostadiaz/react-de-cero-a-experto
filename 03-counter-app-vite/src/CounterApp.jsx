import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
	const [counter, setCounter] = useState(value);
	const handleAdd = () => setCounter((c) => c + 1);
	const handleSubs = () => setCounter((c) => c - 1);
	const handleReset = () => setCounter((c) => value);

	return (
		<>
			<h1>CounterApp</h1>
			<h2>{counter}</h2>

			<button onClick={handleSubs}>-1</button>
			<button aria-label="btn-reset" onClick={handleReset}>
				Reset
			</button>
			<button onClick={handleAdd}>+1</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number,
};

CounterApp.defaultProps = {
	value: 0,
};
