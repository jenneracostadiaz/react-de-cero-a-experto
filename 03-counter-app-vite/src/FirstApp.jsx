import PropTypes from "prop-types";
const newMessage = {
  message: "Hola Mundo",
  title: "Jenner",
};

const suma = (a, b) => a + b;

export const FirstApp = ({ title, subTitle, nombre }) => {
  return (
    <>
      <h1>{title}</h1>
      <p>{subTitle}</p>
      <p>{nombre}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.number,
};

FirstApp.defaultProps = {
  title: "No hay titulo",
  subTitle: 0,
  nombre: "Jenner Acosta",
};
