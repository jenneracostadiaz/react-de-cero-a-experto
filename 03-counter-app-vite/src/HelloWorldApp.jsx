// function App() {
//   return <h1>Hola Mundo</h1>;
// }

// export default App;

export const HelloWorldApp = () => {
  return <h1>HelloWorldApp</h1>;
};
