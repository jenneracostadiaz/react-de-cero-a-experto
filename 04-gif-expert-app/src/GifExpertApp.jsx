import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);
  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories((cat) => [newCategory, ...cat]);
  };
  return (
    <>
      {/* Titulo */}
      <h1>GifExpertApp</h1>
      {/* input */}
      <AddCategory onNewCategory={(value) => onAddCategory(value)} />
      {/* Listado de Gift */}

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}

      {/* Gif Item */}
    </>
  );
};
