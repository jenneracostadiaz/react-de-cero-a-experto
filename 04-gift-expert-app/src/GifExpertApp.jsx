import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);

    const onAddCategory = (newCategory) => {

        if(categories.includes(newCategory))return;

        setCategories( [newCategory, ...categories,] )
        // setCategories( cat => [...categories, 'Valorant'])
    }

    return (
    <>
        <h1>GifExpertApp</h1>

        <AddCategory
            onNewCategory={ onAddCategory }
        />

        { categories.map( (category) => (
                <GifGrid
                    key={ category }
                    category={ category }
                />
            )) 
        }
    </>
)
}
