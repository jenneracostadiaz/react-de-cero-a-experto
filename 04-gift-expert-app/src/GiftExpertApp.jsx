import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GiftExpertApp = () => {
	const [categories, setCategories] = useState(['One Punch']);

	const onAddCategory = (newCategory) => {
		if (categories.includes(newCategory)) return;
		setCategories((categories) => [newCategory, ...categories]);
	};

	return (
		<>
			<h1>GiftExpertApp</h1>
			<AddCategory onNewCategory={onAddCategory} />

			{categories.map((category) => (
				<GifGrid key={category} category={category} />
			))}
		</>
	);
};
