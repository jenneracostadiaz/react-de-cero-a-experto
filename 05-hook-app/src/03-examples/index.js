export * from './LoadignQuote';
export * from './Quote';
export * from './MultipleCustomHooks';