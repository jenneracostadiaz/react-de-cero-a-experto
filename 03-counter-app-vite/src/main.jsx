import React from 'react';
import ReactDOM from 'react-dom/client';

// import { CounterApp } from "./CounterApp";
import { FirstApp } from './FirstApp';
import { CounterApp } from './CounterApp';

import './style.css';

ReactDOM.createRoot(document.getElementById('root')).render(
	<React.StrictMode>
		<CounterApp value={30} />
		{/* <FirstApp
			title="This is a Title"
			subTitle="This is a subtitle"
			nombre="Jenner Acosta"
		/> */}
	</React.StrictMode>
);
