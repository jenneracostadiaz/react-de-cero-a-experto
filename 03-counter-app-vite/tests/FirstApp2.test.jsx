import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => {
	const title = 'Hola, soy Jenner';
	const subTitle = 'Soy un subtitulo';

	test('debe hace match con el snapshot', () => {
		const { container } = render(<FirstApp title={title} />);
		expect(container).toMatchSnapshot();
	});

	test('debe de mostrar el mensaje "Hola, soy Jenner"', () => {
		render(<FirstApp title={title} />);
		expect(screen.getByText(title)).toBeTruthy();
	});

	test('debe de mostrar el titulo en un h1', () => {
		render(<FirstApp title={title} />);
		// expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain();
	});

	test('debe de mostrar el subtitulo enviado por props', () => {
		render(<FirstApp title={title} subTitle={subTitle} />);
		expect(screen.getAllByText(subTitle).length).toBe(1);
	});
});
