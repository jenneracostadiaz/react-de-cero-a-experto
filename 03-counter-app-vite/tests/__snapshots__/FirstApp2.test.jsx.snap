// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hace match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Jenner
  </h1>
  <p>
    No hay subtitulos
  </p>
  <p />
</div>
`;
